src/platform/plugins/shared/unified_search/public/search_bar/create_search_bar.tsx

```diff
@@ -276,6 +276,8 @@ export function createSearchBar({
             filtersForSuggestions={props.filtersForSuggestions}
             prependFilterBar={props.prependFilterBar}
             onESQLDocsFlyoutVisibilityChanged={props.onESQLDocsFlyoutVisibilityChanged}
+            esqlEditorInitialState={props.esqlEditorInitialState}
```

Comment:
nit: If this prop is meant to be used _once_ to set up an internal state store, maybe consider using a callback like `getEsqlEditorInitialState` instead to be more explicit about it.

Reply:
I don't have a strong opinion on it really, but it's how the `restorable-state` package we built works currently. We could look into changing it to a callback if there are strong opinions about it, but I don't really think it's necessary, and should be done in a separate PR if we do since it will affect a number of unrelated components that already use the package.